File: Algorithms/p338_Counting_Bits.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> res(num+1);
        res[0] = 0;
        int m = 1;
        int m2 = 2;
        for(int i=1;i<=num;i++){
            if(i<m2){
                res[i] = res[i-m] + 1;
            }else{
                res[i] = 1;
                m = m2;
                m2 = m*2;
            }
        }
        return res;
    }
};

File: Algorithms/p338_Counting_Bits.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        l = [0]*(num+1)
        m = 1
        m2 = 2
        for i in range(1,num+1):
            if i < m2:
                l[i] = l[i-m] + 1
            else:
                l[i] = 1
                m = m2
                m2 = m*2
        return l
